agent/agent/chain.py

```diff
@@ -67,11 +67,17 @@ def history(self) -> List[dict]:
         )
         past_thoughts = {m.message_id: m.content for m in meta_iter.items}
         for message in iter.items[::-1]:
-            if message.is_user:
-                history_list.append({"role": "user", "content": message.content})
-                history_list.append(
-                    {"role": "assistant", "content": past_thoughts[message.id]}
-                )
+            try:
+                if message.is_user:
+                    history_list.append({"role": "user", "content": message.content})
+                else:  # system message
```

Comment:
Not sure about this being a system message. Might be worth asking @vintrocode about that

Reply:
this is fine for now, i've yet to create a branch to update this repo to the way things work in my [test repo](https://github.com/plastic-labs/bloom-test-fa-24), still blocked on Honcho SDK issues